src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -1107,8 +1107,12 @@ private unsafe SocketError FinishOperationAccept(Internals.SocketAddress remoteS
             int localAddrLength;
             IntPtr remoteAddr;
 
+            bool refAdded = false;
             try
             {
+                _currentSocket.SafeHandle.DangerousAddRef(ref refAdded);
+                IntPtr handle = _currentSocket.SafeHandle.DangerousGetHandle();
```

Comment:
Same comment as earlier about storing the SafeHandle into a local first and operating on that local.